VMWare/vmware-esxi/ingest/parser.yml

```
@@ -0,0 +1,243 @@
+
+          ## This is log pattern to work with change user process. Based on `DEFAULT_PROCESS_LOG_WITH_ADDITIONAL_INFO`
+          ## Example of log:
+          ### info hostd[2099656] [Originator@6876 sub=Vimsvc.ha-eventmgr opID=esxcli-76-ecbf] Event 975279 : User root@127.0.0.1 logged in as pyvmomi Python/3.8.13 (VMkernel; 7.0.3; x86_64)
+          ## Fields output:
+          # "log_lvl": "info"
+          # "process": "hostd"
+          # "pid": 2099656
+          # "source_username": "root"
+          # "ip_address": "127.0.0.1"
+          # "username": "pyvmomi"
+          # "user_agent": " Python/3.8.13 (VMkernel; 7.0.3; x86_64)"  <!NOTE: This field is optional>
+          PROCESS_LOG_CHANGE_USER: '^(?:\s*)(?:%{WORD:log_lvl}) %{DATA:process}\[%{NUMBER:pid}\] \[%{WORD}\@%{NUMBER} %{DATA}\] %{WORD} %{NUMBER} : User %{USERNAME:source_username}@(?:%{IPV6:ip_address}|%{IPV4:ip_address}) %{DATA} as %{WORD:username}%{GREEDYDATA:user_agent}?'
+
+          ## This is default log message pattern that should be used to get log message from events
+          ## Example of log:
+          ### [iscsiCorrelator] 3087813295957us: [vob.iscsi.connection.stopped] iScsi connection 0 stopped for vmhba64:C0:T3
+          DEFAULT_EVENT_LOG: '^(?:\s*)\[%{PROG:correlator_name}\] %{WORD}: \[%{PROG:process}\] %{GREEDYDATA:message}?'
+
+          ## This is default log message pattern that should be used to get log message
+          ## Example of log:
+          ### info hostd[2099655] [Originator@6876 sub=Vimsvc.ha-eventmgr] <LOG_MSG>
+          DEFAULT_PROCESS_LOG: '^(?:\s*)(?:%{WORD:log_lvl}) %{DATA:process}\[%{NUMBER:pid}\] \[%{WORD}\@%{NUMBER}\] %{GREEDYDATA:message}?'
+
+          ## Difference between this one and previous is that meta info can contain various info like `user=<username>`, `Req <Req id>`, `opID=<operation id>` and so on
+          ## Example of log:
+          ### info hostd[2099655] [Originator@6876 sub=Vimsvc.ha-eventmgr opID=233816e0 user=vpxuser Req 06028] <LOG_MSG>
+          DEFAULT_PROCESS_LOG_WITH_ADDITIONAL_INFO: '^(?:\s*)(?:%{WORD:log_lvl}) %{DATA:process}\[%{NUMBER:pid}\] \[%{WORD}\@%{NUMBER} %{DATA}\] %{GREEDYDATA:message}?'
+
+  - name: set_common_fields
+  - name: set_additional_fields
+  - name: set_event_reason
+  - name: set_event_category
+  - name: set_event_type
+
+stages:
+  set_common_fields:
+    actions:
+      # Default fields with predefined values
+      - set:
+          event.kind: 'event'
+          observer.vendor: 'VMware'
+          observer.product: 'ESXi'
+
+  set_additional_fields:
+    actions:
+      # Fields by definition that we get from messages
+      - set:
+          ecs.user.name: '{{parsed_event.result.username}}'
+          ecs.source.ip: '{{parsed_event.result.ip_address}}'
+          ecs.source.port: '{{parsed_event.result.port}}'
+          ecs.source.user.name: '{{parsed_event.result.source_username}}'
+          ecs.user_agent.original: '{{parsed_event.result.user_agent | trim}}'
```

Same fields are already declared in ECS taxonomy. I think they are more suitable:
```suggestion
          user.name: '{{parsed_event.result.username}}'
          source.ip: '{{parsed_event.result.ip_address}}'
          source.port: '{{parsed_event.result.port}}'
          source.user.name: '{{parsed_event.result.source_username}}'
          user_agent.original: '{{parsed_event.result.user_agent | trim}}'
```